src/app/ReadClient.h

```diff
@@ -74,8 +80,18 @@ class ReadClient : public Messaging::ExchangeDelegate
      */
     CHIP_ERROR SendReadRequest(ReadPrepareParams & aReadPrepareParams);
 
+    /**
+     *  Send a subscribe Request.  There can be one Subscribe Request outstanding on a given ReadClient.
+     *  If SendSubscribeRequest returns success, no more subscribe Requests can be sent on this ReadClient
+     *  until the corresponding InteractionModelDelegate::ReadDone call happens with guarantee.
+     *
+     *  @retval #others fail to send subscribe request
+     *  @retval #CHIP_NO_ERROR On success.
+     */
+    CHIP_ERROR SendSubscribeRequest(ReadPrepareParams & aSubscribePrepareParams);
     uint64_t GetAppIdentifier() const { return mAppIdentifier; }
     Messaging::ExchangeContext * GetExchangeContext() const { return mpExchangeCtx; }
+    bool IsSubscription() const { return mInteractionType == InteractionType::Subscribe; };
```

Comment:
```suggestion
    bool IsSubscriptionType() const { return mInteractionType == InteractionType::Subscribe; };
```

Would recommend adding the other counterpart here too `IsReadType()` to be symmetric.

Reply:
Done, thanks